vortex-array/src/stats/mod.rs

```diff
@@ -92,40 +92,32 @@ impl dyn Statistics + '_ {
         &self,
         stat: Stat,
     ) -> Option<U> {
-        self.get(stat)
-            .map(|s| U::try_from(&s).expect("Invalid stats cast"))
+        self.get(stat).and_then(|s| U::try_from(&s).ok())
     }
 
     pub fn get_as_cast<U: NativePType + for<'a> TryFrom<&'a Scalar, Error = VortexError>>(
         &self,
         stat: Stat,
     ) -> Option<U> {
         self.get(stat)
-            .map(|s| {
-                s.cast(&DType::Primitive(U::PTYPE, NonNullable))
-                    .expect("Invalid scalar cast")
-            })
-            .map(|s| U::try_from(&s).expect("Invalid stats cast"))
+            .and_then(|s| s.cast(&DType::Primitive(U::PTYPE, NonNullable)).ok())
+            .and_then(|s| U::try_from(&s).ok())
     }
 
     pub fn compute_as<U: for<'a> TryFrom<&'a Scalar, Error = VortexError>>(
         &self,
         stat: Stat,
     ) -> Option<U> {
-        self.compute(stat)
-            .map(|s| U::try_from(&s).expect("Invalid stats cast"))
+        self.compute(stat).and_then(|s| U::try_from(&s).ok())
     }
 
     pub fn compute_as_cast<U: NativePType + for<'a> TryFrom<&'a Scalar, Error = VortexError>>(
         &self,
         stat: Stat,
     ) -> Option<U> {
         self.compute(stat)
-            .map(|s| {
-                s.cast(&DType::Primitive(U::PTYPE, NonNullable))
-                    .expect("Invalid scalar cast")
-            })
-            .map(|s| U::try_from(&s).expect("Invalid stats cast"))
+            .and_then(|s| s.cast(&DType::Primitive(U::PTYPE, NonNullable)).ok())
+            .and_then(|s| U::try_from(&s).ok())
     }
 
     pub fn compute_min<U: for<'a> TryFrom<&'a Scalar, Error = VortexError>>(&self) -> Option<U> {
```

Comment:
I think this might need a different api. I think until then I would prefer to keep these as is and add an exception